aten/src/ATen/native/IndexingUtils.h

```diff
@@ -0,0 +1,135 @@
+#include <ATen/ExpandUtils.h>
+#include <ATen/native/TensorIterator.h>
+
+namespace at { namespace native {
+
+[[noreturn]]
+static void invalid_mask(const Tensor & self, int64_t idx, const Tensor & mask, int64_t maskIdx) {
+  std::stringstream ss;
+  ss << "The shape of the mask " << mask.sizes() << " at index " << maskIdx;
+  ss << " does not match the shape of the indexed tensor " << self.sizes();
+  ss << " at index " << idx;
+  AT_INDEX_ERROR(ss.str());
```

Comment:
FYI, you don't need to construct the stringstream manuallyl; `AT_INDEX_ERROR` takes any number of comma arguments and will cat them together using a stringstream. Also, this macro got a new name; it's `TORCH_CHECK_INDEX(false, ...)` now (No need to fix either of these issues unless you happen to be addressing other fixes on this PR).